romancal/associations/tests/test_skycell_asn.py

```diff
@@ -30,3 +31,180 @@ def test_parse_visitID():
     assert visitid_parts["Segment"] == "003"
     assert visitid_parts["Observation"] == "004"
     assert visitid_parts["Visit"] == "005"
+
+
+@pytest.fixture
+def sample_filelist():
+    return [
+        "r0000101002003004005_0001_wfi10_cal.asdf",
+        "r0000101002003004005_0002_wfi10_cal.asdf",
+        "r0000101002003004006_0001_wfi10_cal.asdf",
+    ]
+
+
+def test_create_groups_full(sample_filelist):
+    groups = skycell_asn._create_groups(sample_filelist, "full")
+    assert "full" in groups
+    assert set(groups["full"]) == set(sample_filelist)
+
+
+def test_create_groups_visit(sample_filelist):
+    groups = skycell_asn._create_groups(sample_filelist, "visit")
+    # Should group by visit id (first 19 chars after 'r')
+    assert all(isinstance(v, list) for v in groups.values())
+    assert len(groups) == 2  # two unique visit IDs in sample_filelist
+
+
+def test_create_groups_pass(sample_filelist):
+    groups = skycell_asn._create_groups(sample_filelist, "pass")
+    assert all(isinstance(v, list) for v in groups.values())
+    # Should group by pass key (positions 5:10)
```

Comment:
I need to read further in the PR but 'program' is also relevant here, so it's really positions 0:10.